app/src/main/java/org/pesmypetcare/mypetcare/services/CommunityAdapter.java

```diff
@@ -308,28 +367,105 @@ public void updatePost(User user, Post post, String newText) {
-    public void likePost(String likerName, String authorName, DateTime creationDate, String forumName,
-                         String groupName) throws PostNotFoundException {
-        /*ExecutorService executorService = Executors.newSingleThreadExecutor();
+    public void likePost(User user, Post post) {
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
         executorService.execute(() -> {
             try {
-                ServiceLocator.getInstance().getForumManagerClient().likePost(...);
+                ServiceLocator.getInstance().getForumManagerClient().likeMessage(user.getToken(), user.getUsername(),
+                    post.getForum().getGroup().getName(), post.getForum().getName(), post.getUsername(),
+                    post.getCreationDate().toString(), true);
             } catch (MyPetCareException e) {
                 e.printStackTrace();
             }
-        });*/
+        });
+
+        executorService.shutdown();
     }
 
     @Override
     public void unlikePost(User user, Post post) {
-        // Not implemented yet
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        executorService.execute(() -> {
+            try {
+                ServiceLocator.getInstance().getForumManagerClient().likeMessage(user.getToken(), user.getUsername(),
+                    post.getForum().getGroup().getName(), post.getForum().getName(), post.getUsername(),
+                    post.getCreationDate().toString(), false);
+            } catch (MyPetCareException e) {
+                e.printStackTrace();
+            }
+        });
+
+        executorService.shutdown();
     }
 
     @Override
     public void reportPost(User user, Post post, String reportMessage) {
         // Not implemented yet
     }
 
+    @Override
+    public void addPostImage(User user, Post post, Bitmap image) throws PostNotFoundException {
+        // Not implemented yet
+    }
+
+    @Override
+    public void deletePostImage(User user, Post post) throws PostNotFoundException {
+        // Not implemented yet
+    }
+
+    @Override
+    public void addGroupImage(User user, Group group, Bitmap image) throws MyPetCareException {
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        executorService.execute(() -> {
+            byte[] imageBytes = ImageManager.getImageBytes(image);
+            try {
+                ServiceLocator.getInstance().getGroupManagerClient().updateGroupIcon(user.getToken(), group.getName(),
+                    imageBytes);
+            } catch (MyPetCareException e) {
+                e.printStackTrace();
+            }
+        });
+
+        executorService.shutdown();
+    }
+
+    @Override
+    public void deleteGroupImage(User user, Group group) throws GroupNotFoundException {
+        // Not implemented yet
+    }
+
+    @Override
+    public byte[] getPostImage(User user, Post post, MessageDisplay messageDisplay) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Return an empty array rather than null.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5529921)